tests/e2e/vmlistener_test.go

```diff
@@ -59,8 +59,10 @@ func (s *VMListenerTestParams) SetUpTest(c *C) {
 }
 
 func (s *VMListenerTestParams) TearDownTest(c *C) {
-	// Note: no need to remove container, it is already removed as
-	// vm got killed and `--restart=always` flag is not added
+	if dockercli.IsContainerExist(s.dockerHostIP, s.containerName) {
```

Comment:
Shouldn't SetupSuite() use the inputparams getEndPoint*() functions to get the dockerHostIP()? Same for the ESX IP.